components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -716,4 +724,27 @@ void ESP8266Interface::proc_oob_evnt()
         _esp.bg_process_oob(ESP8266_RECV_TIMEOUT, true);
 }
 
+nsapi_error_t ESP8266Interface::_conn_status_to_error()
+{
+    nsapi_error_t ret;
+
+    _esp.bg_process_oob(ESP8266_RECV_TIMEOUT, true);
+
+    switch (_conn_stat) {
+        case NSAPI_STATUS_DISCONNECTED:
+            ret = NSAPI_ERROR_NO_CONNECTION;
+            break;
+        case NSAPI_STATUS_CONNECTING:
+            ret = NSAPI_ERROR_ALREADY;
+            break;
+        case NSAPI_STATUS_GLOBAL_UP:
+            ret = NSAPI_ERROR_IS_CONNECTED;
+            break;
+        default:
```

Comment:
Maybe, but what is a watchdog reset if not a device error.

Reply:
Right, I actually wonder why getting a watchdog reset causes the connection status to be UNSUPPORTED... Seems like there is just no better status for it and it will likely be set to something meaningful after the reset takes place. 
I agree - it is better leave the code as is :)

Reply:
Got an idea that it would be better to call MBED_ERROR if the modem crashes and watchdog reset occurs. Better to reveal the crash as soon as possible.

Reply:
Fully agree. It is not relevant for this PR, so I added it to internal JIRA ticket ONME-4141, so we don't forget.

Reply:
Created an issue #9425